SGoopas/Assets/Scripts/Game/2D/EndTrigger.cs

```diff
@@ -4,8 +4,29 @@
 
 public class EndTrigger : MonoBehaviour
 {
+    private Animator anim;
+    //private GameObject player;
+
+    private void Start()
+    {
+        anim = GetComponent<Animator>();
+        //player = MainObjectContainer.Instance.Player2D;
+    }
+
     void OnTriggerEnter2D(Collider2D other)
     {
+        ExitHandler.TriggerExit();
+        anim.SetBool("exit", true);
+        StartCoroutine(ExitLevel());

```

Comment:
This is what's causing the multiple level loading, I think. As the player swirls around and collides with the trigger, this coroutine starts a couple times